src/peft/tuners/lora/model.py

```diff
@@ -791,3 +793,30 @@ def unload(self) -> torch.nn.Module:
         model.
         """
         return self._unload_and_optionally_merge(merge=False)
+
+    def subtract_pissa_init(self, output_state_dict: str, adapter_name: str = "pissa_init", kwargs=None):
+        """
+        This function can calculate the updates of the PiSSA by comparing the parameters of the PiSSA adapter
+        in `output_state_dict` with the initial values of PiSSA in `adapter_name`, thus converting PiSSA to LoRA.
+        """
+        pissa_init_state_dict = get_peft_model_state_dict(
+            self,
+            state_dict=kwargs.get("state_dict", None),
+            adapter_name=adapter_name,
+        )
+        tensors_lora = {}
+        for name in output_state_dict.keys():
+            ## W = W^{res} + A_0 \times B_0,
+            ## W + \Delta W = W^{res} + A \times B,
+            ## \Delta W = A \times B - A_0 \times B_0 = [A | A_0] \times [B | -B_0]^T = A'B'.
+            if "lora_A" in name:
+                tensors_lora[name] = torch.cat(
+                    [output_state_dict[name], pissa_init_state_dict[".".join(name.split(".")[1:])]], dim=0
+                )
+            else:
```

Comment:
Might not be necessary but would be safer, WDYT?

```suggestion
            elif "lora_B" in name:
```